x/nexus/keeper/lockable_asset.go

```diff
@@ -29,6 +21,38 @@ type lockableAsset struct {
 	bank     types.BankKeeper
 }
 
+// NewLockableAsset creates a new lockable asset from a nexus registered asset name.
+func (k Keeper) NewLockableAsset(ctx sdk.Context, ibc types.IBCKeeper, bank types.BankKeeper, coin sdk.Coin) (exported.LockableAsset, error) {
+	return newLockableAsset(ctx, k, ibc, bank, coin)
+}
+
+// NewLockableAssetFromCosmosCoin creates a new lockable asset from a Cosmos x/bank Coin (e.g. uaxl, ICS20 coin).
+func (k Keeper) NewLockableAssetFromCosmosCoin(ctx sdk.Context, ibc types.IBCKeeper, bank types.BankKeeper, coin sdk.Coin) (exported.LockableAsset, error) {
+	if isIBCDenom(coin.GetDenom()) {
+		denomTrace, err := ibc.ParseIBCDenom(ctx, coin.GetDenom())
+		if err != nil {
+			return lockableAsset{}, err
+		}
+
+		coin = sdk.NewCoin(denomTrace.GetBaseDenom(), coin.Amount)
+	}
+
+	asset, err := newLockableAsset(ctx, k, ibc, bank, coin)
+	if err != nil {
+		return lockableAsset{}, err
+	}
+
+	if convertedCoin, err := asset.getCoin(ctx); err != nil {
+		return lockableAsset{}, err
+	} else if convertedCoin != coin {
```

Comment:
I'd like to keep the ICS20 handling and checks separate and explicit, so it's clear when they're actually needed. Alternative solutions: Define this wrapper/constructor only in `axelarnet` module that is more ICS20 aware than nexus